engines/director/lingo/xlibs/popupmenuxobj.cpp

```diff
@@ -149,26 +152,72 @@ void PopUpMenuXObj::close(int type) {
 }
 
 
-PopUpMenuXObject::PopUpMenuXObject(ObjectType ObjectType) :Object<PopUpMenuXObject>("PopMenu") {
+PopUpMenuXObject::PopUpMenuXObject(ObjectType ObjectType) : Object<PopUpMenuXObject>("PopMenu") {
 	_objType = ObjectType;
 }
 
 void PopUpMenuXObj::m_new(int nargs) {
-	g_lingo->printSTUBWithArglist("PopUpMenuXObj::m_new", nargs);
-	g_lingo->dropStack(nargs);
+	PopUpMenuXObject *me = static_cast<PopUpMenuXObject *>(g_lingo->_state->me.u.obj);
+	Datum menuId = g_lingo->pop();
+	Datum menuList = g_lingo->pop();
+
+	new Graphics::MacPopUp(menuId.asInt(), g_director->_wm->getScreenBounds(), g_director->_wm, menuList.u.s->c_str());
```

Comment:
`menuList.u.s->c_str()` is a direct access to the union. If, for whatever reason, you do not get a string in there, you will have bad pointer and a SIGSEGV crash. Please use `asString()` method and in general, use `as*()` methods because those perform the needed conversions.

For readability, as it was mentioned several reviews back, use the explicit data types, e.g. instead of the current code, write:

```C++
int menuId = g_lingo->pop().asInt();
Common::String menuList = g_lingo->pop().asString();
```

And then use these directly.